src/backend/marsha/core/permissions.py

```diff
@@ -56,102 +58,74 @@ def has_permission(self, request, view):
-        request : Type[django.http.request.HttpRequest]
-            The request that holds the authenticated user
-        view : Type[restframework.viewsets or restframework.views]
-            The API view for which permissions are being checked
-        obj: Type[models.Model]
-            The object for which object permissions are being checked
-
-        Raises
-        ------
-        PermissionDenied
-            Raised if the request is not authorized
-
-        Returns
-        -------
-        None
-            The response of this method is ignored by its caller in restframework (we delegate
-            checking if the user is Admin to the parent class `IsAdminUser`)
-
-        """
-        # Users authentified via LTI are identified by a TokenUser with the
-        # resource_link_id as user ID.
-        if str(self.get_resource_id(obj)) == request.user.id:
-            return True
-
-        return False
-
 
-class IsResourceInstructor(BaseResourcePermission):
+class IsTokenInstructor(BaseTokenRolePermission):
     """Class dedicated to instructor users."""
 
     role = INSTRUCTOR
 
 
-class IsResourceAdmin(BaseResourcePermission):
+class IsTokenAdmin(BaseTokenRolePermission):
     """Class dedicated to administrator users."""
 
     role = ADMINISTRATOR
 
 
-class BaseVideoRelatedPermission(BaseResourcePermission):
-    """A custom permission class for JWT Tokens related to objects linked to a video.
-
-    These permissions build on the `IsAdminUser` class but grants additional specific accesses
-    to users authenticated with a JWT token built from the LTI resource link id ie related to a
-    TokenUser as defined in `rest_framework_simplejwt`.
+class IsTokenResourceRouteObject(permissions.BasePermission):
+    """
+    Base permission class for JWT Tokens related to a resource object.
 
+    These permissions grants access to users authenticated with a JWT token built from a
+    resource ie related to a TokenUser as defined in `rest_framework_simplejwt`.
     """
 
-    def get_resource_id(self, obj):
-        """Get the video id to check that it matches the JWT Token.
+    def has_permission(self, request, view):
+        """
+        Add a check to allow the request if the JWT resource matches the object in the url path.
 
         Parameters
         ----------
-        obj : Type[models.Model]
-            The  object for which object permissions are being checked
+        request : Type[django.http.request.HttpRequest]
+            The request that holds the authenticated user
+        view : Type[restframework.viewsets or restframework.views]
+            The API view for which permissions are being checked
 
         Returns
         -------
-        string
-            The id of the video linked to the object passed as obj
-
+        boolean
+            True if the request is authorized, False otherwise
         """
-        return obj.video.id
+        return view.get_object_pk() == request.user.id
```

Comment:
Should we add a warning here that request.user is not a user? :fearful: 
I find this part really confusing because even though I know it, my brain tells me it's a user each time I read it. 
Don't hesitate to repeat it in the comments.